client/src/components/Game/Timer.js

```diff
@@ -0,0 +1,136 @@
+import { withRouter } from 'react-router';
+import { compose } from 'recompose';
+
+class Timer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+
+    }
+  }
+
+  async componentDidMount() {
+    const { id } = this.props.match.params;
+    const game = this.props.firebase.game(id);
+
+    game.get()
+    .then(docSnapshot => {
+      if(docSnapshot.exists) {
+        this.unsubscribe = game.onSnapshot((snapshot) => {
+          if(snapshot.data().gameStartTime !== null && snapshot.data().currentPlayerIndex === 0) {
+            if(snapshot.data().currentPlayerIndex === 0) {
+              this.setState({
+                startTime: snapshot.data().gameStartTime.seconds,
+                timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+              })
+            } else {
+              this.setState({
+                startTime: Math.floor(new Date().getTime() / 1000),
+                timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+              })
+            }
+            this.setState({
+              timeTurnWillEnd: (this.state.timePerTurnInSeconds + this.state.startTime),
+            })
+            this.setState({
+              days: this.calculateTime('days'),
+              hours: this.calculateTime('hours'),
+              minutes: this.calculateTime('minutes'),
+              seconds: this.calculateTime('seconds')
+            })
+          }
+        })
+      }
+    })
+
+    this.myInterval = setInterval(async () => {
+      const { days, hours, minutes, seconds } = this.state
+      const gameRef = await this.props.firebase.game(id).get();
+
+      if (seconds > 0) {
+        this.setState({
+          seconds: this.calculateTime('seconds')
+        })
+      }
+      if (seconds === 0) {
+        if (minutes === 0 && hours === 0 && days === 0) {
+          this.sendEmail(gameRef, id);
+          game.set({
+            currentPlayerIndex: gameRef.data().currentPlayerIndex + 1,
+          }, { merge: true })
+          clearInterval(this.myInterval)
+        } else {
+          this.setState({
+            days: this.calculateTime('days'),
+            hours: this.calculateTime('hours'),
+            minutes: this.calculateTime('minutes'),
+            seconds: this.calculateTime('seconds')
+          })
+        }
+      }
+    }, 1000)
+  }
+
+  calculateTime(unitOfTime) {
+    switch(unitOfTime) {
+      case 'days':
+        return Math.floor((((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) / 60)/ 24);
+      case 'hours':
+        return Math.floor(((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) / 60) % 24;
+      case 'minutes':
+        return Math.floor((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) % 60;
+      case 'seconds':
+        return (this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) % 60;
```

Comment:
nit: for readability's sake, probably move repetitive computation outside of switch-case